src/boson_sampling_utilities/permanent_calculators/ClassicPermanentCalculator.py

```diff
@@ -0,0 +1,79 @@
+__author__ = "Tomasz Rybotycki"
+
+"""
+    This class is used to calculate permanent of effective scattering matrix. It first generates the matrix, and then
+    calculates the permanent via standard means.
+"""
+
+from typing import List, Optional
+
+from numpy import array, asarray, complex128, int64, ndarray
+
+from src.boson_sampling_utilities.Boson_Sampling_Utilities import EffectiveScatteringMatrixCalculator
+from src.boson_sampling_utilities.permanent_calculators.BSPermanentCalculatorInterface import \
+    BSPermanentCalculatorInterface
+
+
+class ClassicPermanentCalculator(BSPermanentCalculatorInterface):
```

Comment:
That's smart. I actually didn't know how to proceed here, since the interface is not the place to implement anything, and creating another class that would be used as a base seemed like overdoing it. During the refactoring I can see why it is NOT. Thank you for that.